### PR TITLE
Fix #8088 - Restructure source connections & clients

### LINKED ISSUE #8088: Split connections into get_connection and test_connection files
The [connections](https://github.com/open-metadata/OpenMetadata/blob/main/ingestion/src/metadata/utils/connections.py) module has grown too big and it's hard to pinpoint what we are doing where.

Let's split the logic into two files, one with get_connection and another with test_connections.

Also, instead of having a module `utils`, let's move that to a module `connections`.

### PR DESCRIPTION
**Pull request by pmbrull**
### Describe your changes :

Fix #8088

After updating the ingestion module structure, we can move the get_connection and test_connection methods close to their source module. The same can be applied for queries and clients.

The sql_queries module is still hanging around. We'll clean that up once we start implementing the performance improvements.

#
### Type of change :

- [x] Bug fix
- [x] Improvement
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: ingestion/setup.py [hunk 1]
```diff
@@ -45,7 +45,9 @@ def get_long_description():
     "boto3~=1.19.12",
     "botocore==1.22.12",
     "avro-python3==1.10.2",
-    "grpcio-tools==1.48.2",
+    # https://github.com/grpc/grpc/issues/30843#issuecomment-1303816925
```
Comment (pmbrull) [inline ingestion/setup.py:48]: taking the chance to update some versions to make our `grpcio` dependencies compatible with dagster base requirements

**Comment by harshach** [comment]
@pmbrull can we also make sure we update all the ingestion dependencies to latest available versions

File: ingestion/src/metadata/clients/aws_client.py [hunk 2]
```diff
@@ -91,20 +83,20 @@ def get_resource(self, service_name: str) -> Any:
...
 
-    def get_sagemaker_client(self) -> SageMakerClient:
-        return SageMakerClient(self.get_client("sagemaker"))
+    def get_sagemaker_client(self):
+        return self.get_client("sagemaker")
 
-    def get_kinesis_client(self) -> KinesisClient:
-        return KinesisClient(self.get_client("kinesis"))
+    def get_kinesis_client(self):
+        return self.get_client("kinesis")
```
Comment (harshach) [inline ingestion/src/metadata/clients/aws_client.py:99]: shall we make this as enum or service.json enums? instead of strings

**Comment by pmbrull** [comment]
> @pmbrull can we also make sure we update all the ingestion dependencies to latest available versions

Let's handle this as another ticket, don't want to add more noise here

**Comment by pmbrull** [comment]
@ayush-shah @TeddyCr not sure why the `test_profiler_partitions` tests are failing 🤔 running them from Pycharm works, but through the Makefile they die. I could use a hand here. Thanks!

**Review by ulixius9 — APPROVED** [review]

**Pull request merged**